src/Admin/UserManager.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Class UserManager.
+ *
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Service;
+
+/**
+ * Class UserManager
+ *
+ * @since 1.6.0
+ */
+final class UserManager implements Service {
+
+	/**
+	 * Custom user capability allowing dev tools
+	 *
+	 * @var string
+	 */
+	const VALIDATE_CAP = 'amp_validate';
+
```

Comment:
No users currently have this capability, right?

We'll want to utilize `map_meta_cap` actually for this (linking to `manage_options`) rather than to grant the capability directly to users or to a given role.

Reply:
@westonruter After revisiting this PR and also reviewing https://github.com/ampproject/amp-wp/issues/2673, I think would make sense to completely separate the `amp_dev_tools_enabled` option from the capability. A user will be able to set `amp_dev_tools_enabled` according to their preference, while an administrator would have some other way of granting the capability to non-administrators. This might involve a separate user meta field, but I think that would be outside of the scope of this PR after all. Does this sound right to you?

In sum:

1. Administrators can grant users access to dev tools (not in this PR, maybe in #2673 instead)
2. Separately, users with access to dev tools can turn them off or on according to their preference

Reply:
So then the capability check to update this field would be:

```php
current_user_can( 'manage_options' ) && current_user_can( 'edit_user', $user_id )
```

Is that right?